lib/twitter/accounts.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Twitter.Accounts do
+  @moduledoc """
+  The Accounts context.
+  """
+  alias Twitter.Repo
+  alias Twitter.Accounts.User
+
+  @doc """
+  create user: validate parameters and insert in the db
+  """
+  @spec create_user(map()) ::
+          {:ok, User.t()} | {:error, Ecto.Changeset.t(User.t())}
+  def create_user(attrs \\ %{}) do
+    %User{}
+    |> User.registration_changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  get_user: returns user by id
+  """
+  @spec get_user(non_neg_integer() | String.t()) :: User.t() | nil
+  def get_user(id) do
+    Repo.get(User, id)
+  end
+
+  @doc """
+  returns user by id or raises Ecto.NoResultsError if not found
+  """
+  @spec get_user!(non_neg_integer() | String.t()) :: User.t()
+  # has error handling
+  def get_user!(id) do
+    Repo.get!(User, id)
+  end
+
+  @doc """
+  returns a single user by any parameter
+  """
+  @spec get_user_by(map()) :: User.t() | nil
+  def get_user_by(params) do
+    Repo.get_by(User, params)
```

Comment:
Not all params are guaranteed to map to a unique user, so this function is probably too generic.
Actually unique fields are often limited, so it is best to only make specialized functions for real use cases, e.g. get by email / get by username...

Reply:
yes! makes much sense. 
Since I'm not using it anywhere for now I will remove it, but I can see how it could be useful if I do a users search and filter functions.